tests/std/tests/GH_004597_self_swap/test.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <utility>
+
+struct swap_counter {
+    unsigned int* pcnt_ = nullptr;
+
+    friend _CONSTEXPR20 void swap(swap_counter& lhs, swap_counter& rhs) noexcept {
+        std::swap(lhs.pcnt_, rhs.pcnt_);
+        if (lhs.pcnt_ != nullptr) {
+            ++(*lhs.pcnt_);

```

Comment:
The parentheses here are unnecessary - there's no other way that the `++` and `*` operators could possibly bind. Occurs below.

We generally avoid unnecessary parentheses, to make the necessary ones stand out more. We do make exceptions for certain visually confusing cases - sometimes conditional operators, and always when bitwise operators are mixed with anything else.